### ISSUE TITLE
put_pixel( region, data ) available on video memory (feature request)

### ISSUE THREAD

**Issue by JohnMeadow1** [godotengine/godot#7146]
**Operating system or device - Godot version:**
Godot 2.1

**Issue description** (what happened, and what was expected):
I'm using pixel per pixel calculations and texture modifications at runtime. 
At the moment Godot does not allow pixel/region texture updates on video memory. You can only update entire texture in video memory at once. 
Before switching  to Godot I was doing this in C++/ OpenGL. Where entire environment was destroyable pixel per pixel. Doing the same in Godot is possible only to certain texture size, after that performance happens.
This kind of problem requires you to keep a copy of the textures in Ram and update any changes to Vram.
This can look like this: 
```
data = self.get_texture().get_data()
imageTexture = ImageTexture.new()
imageTexture.create(width, height, self.get_texture().get_format(), 0)    
imageTexture.set_data(data)
data.put_pixel( position.x, position.y, Color(1,0,0,1), 0 )
imageTexture.set_data(data)
```
If you need to run following code every 5-20 frames it becomes problematic.  
```
data.put_pixel( position.x, position.y, Color(1,0,0,1), 0 )
imageTexture.set_data(data)
```
It is possible to update texture once per second or so, but if the textures are 2048x2048 then it is still a bottleneck.

Being able to call functions like the ones defined for Image class, but straight on video memory would allow new possibilities for the engine. 
Especially:
```
blit_rect ( Image src, Rect2 src_rect, Vector2 dest=0 )
Color 	get_pixel ( int x, int y, int mipmap_level=0 )
Image 	get_rect ( Rect2 area=0 )
void 	put_pixel ( int x, int y, Color color, int mipmap_level=0 )
```
I was wondering how difficult would it be to add this feature?

**Comment by Noshyaar** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
As there was no answer to the above request, we're closing this issue as inactive.
If you can still reproduce it, please open a new issue against a currently supported release of Godot.
